pkg/controller/handlers/knowledgeset/knowledgeset.go

```diff
@@ -149,7 +150,7 @@ func (h *Handler) CreateWorkspace(req router.Request, _ router.Response) error {
 
 func (h *Handler) Cleanup(req router.Request, _ router.Response) error {
 	ks := req.Object.(*v1.KnowledgeSet)
-	if ks.Status.ThreadName == "" || ks.Status.EmptyDatasetDeleted || (ks.DeletionTimestamp.IsZero() && ks.Status.HasContent) {
+	if ks.Status.ThreadName == "" || !ks.Status.DatasetCreated || ks.Status.EmptyDatasetDeleted || (ks.DeletionTimestamp.IsZero() && ks.Status.HasContent) {
```

Comment:
@thedadams I am not sure if `EmptyDatasetDeleted` is still needed here.

Reply:
It seems like DatasetDeleted and DatasetCreated are the same thing, and `DatasetCreated` is a friendlier name, so I am good with getting rid of the other one.

Reply:
sounds good. The EmptyDatasetDeleted was not really preventing it from being deleted when it is first created so I will just remove that and use the new field.